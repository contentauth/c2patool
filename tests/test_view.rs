mod test_utils;

use insta_cmd::assert_cmd_snapshot;
use test_utils::{cmd, test_img_path};

#[test]
fn test_view_manifest() {
    assert_cmd_snapshot!(cmd().arg("view").arg("manifest").arg(test_img_path()));
}

// TODO: remove variable changing info from output
#[test]
fn test_view_manifest_debug() {
    assert_cmd_snapshot!(cmd()
        .arg("view")
        .arg("manifest")
        .arg("--debug")
        .arg(test_img_path()));
}

// TODO: remove variable changing info from output
#[test]
fn test_view_ingredient() {
    assert_cmd_snapshot!(cmd().arg("view").arg("ingredient").arg(test_img_path()));
}

#[test]
fn test_view_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("infop").arg(test_img_path()));
}

#[test]
fn test_view_manifest_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("tree").arg(test_img_path()));
}

// TODO: remove variable changing info from output
#[test]
fn test_view_manifest_certs() {
    assert_cmd_snapshot!(cmd().arg("view").arg("certs").arg(test_img_path()));
}
