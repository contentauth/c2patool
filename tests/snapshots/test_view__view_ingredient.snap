---
source: tests/test_view.rs
info:
  program: c2patool
  args:
    - view
    - ingredient
    - /Users/nicky/Documents/repos/c2patool/tests/fixtures/C.jpg
---
success: true
exit_code: 0
----- stdout -----
{
  "title": "C.jpg",
  "format": "image/jpeg",
  "instance_id": [XMP_ID]",
  "provenance": "[JUMBF_URI]",
  "thumbnail": {
    "format": "image/jpeg",
    "identifier": "[JUMBF_URI]",
    "hash": "BkI+QPdyZ1fEXETg2uOBTpM5wmlwx+erq7a8KnDTOt4="
  },
  "relationship": "componentOf",
  "active_manifest": "[MANIFEST_URN]",
  "manifest_data": {
    "format": "application/c2pa",
    "identifier": "manifest_data.c2pa"
  }
}

----- stderr -----
